flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java

```diff
@@ -115,4 +127,35 @@ public static boolean isDeclarativeResourceManagementEnabled(Configuration confi
         return configuration.get(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT)
                 && !System.getProperties().containsKey("flink.tests.disable-declarative");
     }
+
+    /** The mode of how to handle user code attempting to exit JVM. */
+    public enum UserSystemExitMode {
+        DISABLED("Flink is not monitoring or intercepting calls to System.exit()"),
+        LOG("Log exit attempt with stack trace but still allowing exit to be performed"),
+        THROW("Throw exception when exit is attempted disallowing JVM termination");
+
+        private final String description;
+
+        UserSystemExitMode(String description) {
+            this.description = description;
+        }
+
+        public static Description getConfigDescription() {
+            Description.DescriptionBuilder builder = Description.builder();
+            List<TextElement> modeDescriptions =
+                    new ArrayList<>(UserSystemExitMode.values().length);
+            builder.text(
+                    "Flag to check user code exiting system by terminating JVM (e.g., System.exit())");
+            for (UserSystemExitMode mode : UserSystemExitMode.values()) {
+                modeDescriptions.add(
+                        text(String.format("%s - %s", mode.name(), mode.getDescription())));
+            }
+            builder.list(modeDescriptions.toArray(new TextElement[modeDescriptions.size()]));
```

Comment:
```suggestion
            builder.list(modeDescriptions.toArray(new TextElement[modeDescriptions.size()]));
            builder.text(
                    "This configuration value might change the behavior of %s, when configured to %s and System.exit() gets called in user code.",
                    code(HALT_ON_FATAL_ERROR.key()), code(THROW.name()));
```

Reply:
Two questions:
1. Above header explanation, `Flag to check user code exiting system by terminating JVM` is added already to indicate it's for user code. Do you think one more emphasis at the end like you described?
2. In your suggested documentation, I think `HALT_ON_FATAL_ERROR` should be `INTERCEPT_USER_SYSTEM_EXIT`.

Reply:
Mh, that you are asking these questions makes me doubt whether this change is necessary.
Let me change my suggestion to
```java
            builder.list(modeDescriptions.toArray(new TextElement[modeDescriptions.size()]));
            builder.linebreak();
            builder.text(
                    "Note that this configuration option can interfere with %s: In intercepted user-code, a call to System.exit() will not cause the JVM to halt, when %s is configured.",
                    code(HALT_ON_FATAL_ERROR.key()), code(THROW.name()));
```
Is it clearer now?

I'm also happy to not apply this suggestion. I believe we are optimizing the last 1% out of two rarely used expert features.

Reply:
>In your suggested documentation, I think HALT_ON_FATAL_ERROR should be INTERCEPT_USER_SYSTEM_EXIT.

This part is obviously due to my misread of your comment. The new explanation seems clearer.

>I'm also happy to not apply this suggestion. I believe we are optimizing the last 1% out of two rarely used expert features.

Agreed but maybe adding this explanation may help. Added.